ccx/api/datastore_firewall.go

```diff
@@ -0,0 +1,227 @@
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Authorization", token)
+	client := &http.Client{Timeout: ccx.DefaultTimeout}
+
+	res, err := client.Do(req)
+	if err != nil {
+		return errors.Join(ccx.RequestSendingErr, err)
+	}
+
+	if res.StatusCode == http.StatusBadRequest {
+		return lib.ErrorFromErrorResponse(res.Body)
+	}
+
+	if res.StatusCode != http.StatusCreated {
+		return fmt.Errorf("%w: status = %d", lib.ErrorFromErrorResponse(res.Body), res.StatusCode)
+	}
+
+	return nil
+}
+
+func (svc *DatastoreService) CreateFirewallRules(ctx context.Context, storeID string, firewalls []ccx.FirewallRule) error {
+	limiter := make(chan bool, 10)
+
+	var eg errgroup.Group
+
+	for _, f := range firewalls {
+		limiter <- true
+		f := f
+		eg.Go(func() error {
+			defer func() { <-limiter }()
+			err := svc.CreateFirewallRule(ctx, storeID, f)
+
+			if err != nil {
+				return fmt.Errorf("creating rule (source=%s, description=%s): %w", f.Source, f.Description, err)
+			}
+
+			return nil
+		})
+	}
+
+	if err := eg.Wait(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (svc *DatastoreService) DeleteFirewallRule(ctx context.Context, storeID string, firewall ccx.FirewallRule) error {
+	var b bytes.Buffer
+	if err := json.NewEncoder(&b).Encode(firewall); err != nil {
+		return errors.Join(ccx.RequestEncodingErr, err)
+	}
+
+	url := svc.baseURL + "/api/firewall/api/v1/firewall/" + storeID
+	req, err := http.NewRequest(http.MethodDelete, url, &b)
+	if err != nil {
+		return errors.Join(ccx.RequestInitializationErr, err)
+	}
+
+	token, err := svc.auth.Auth(ctx)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Authorization", token)
+	client := &http.Client{Timeout: ccx.DefaultTimeout}
+
+	res, err := client.Do(req)
+	if err != nil {
+		return errors.Join(ccx.RequestSendingErr, err)
+	}
+
+	if res.StatusCode == http.StatusBadRequest {
+		return lib.ErrorFromErrorResponse(res.Body)
+	}
+
+	if res.StatusCode != http.StatusAccepted {
+		return fmt.Errorf("%w: status = %d", lib.ErrorFromErrorResponse(res.Body), res.StatusCode)
+	}
+
+	return nil
+}
+
+func (svc *DatastoreService) DeleteFirewallRules(ctx context.Context, storeID string, firewalls []ccx.FirewallRule) error {
+	limiter := make(chan bool, 10)
+
+	var eg errgroup.Group
+
+	for _, f := range firewalls {
+		limiter <- true
+		f := f
+		eg.Go(func() error {
+			defer func() { <-limiter }()
+			err := svc.DeleteFirewallRule(ctx, storeID, f)
+
+			if err != nil {
+				return fmt.Errorf("deleting rule (source=%s, description=%s): %w", f.Source, f.Description, err)
+			}
+
+			return nil
+		})
+	}
+
+	if err := eg.Wait(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (svc *DatastoreService) SetFirewallRules(ctx context.Context, storeID string, firewalls []ccx.FirewallRule) error {
+	have, err := svc.GetFirewallRules(ctx, storeID)
+	if err != nil {
+		return fmt.Errorf("getting firewalls: %w", err)
+	}
+
+	create, del := firewallsDiff(have, firewalls)
```

Comment:
FYI, these are merged in terraform config, diffed and split here, then merged in datastores, then diffed and split in deployer.